extensions/prototype-content-with-issues-table/get_usage.R

```diff
@@ -0,0 +1,58 @@
+library(connectapi)
+
+# This file contains functions that ultimately will more likely be part of
+# connectapi. As such, I'm not using dplyr or pipes here.
+
+NA_datetime_ <- vctrs::new_datetime(NA_real_, tzone = "UTC")
+NA_list_ <- list(list())
+
+usage_dtype <- tibble::tibble(
+  "id" = NA_integer_,
+  "user_guid" = NA_character_,
+  "content_guid" = NA_character_,
+  "timestamp" = NA_datetime_,
+  "data" = NA_list_
+)
+
+# A rough implementation of how a new firehose usage function would work in
+# `connectapi`.
+get_usage_firehose <- function(client, from = NULL, to = NULL) {
+  usage_raw <- client$GET(
+    connectapi:::unversioned_url("instrumentation", "content", "hits"),
+    query = list(
+      from = from,
+      to = to
+    )
+  )
+
+  # FIXME for connectapi: This is slow, it's where most of the slowness is with
+  # the new endpoint.
+  usage_parsed <- connectapi:::parse_connectapi_typed(usage_raw, usage_dtype)
+
+  usage_parsed[c("user_guid", "content_guid", "timestamp")]
+}
+
+get_usage_legacy <- function(client, from = NULL, to = NULL) {
+  shiny_usage <- get_usage_shiny(client, limit = Inf, from = from, to = to)
+  shiny_usage_cols <- shiny_usage[c("user_guid", "content_guid")]
+  shiny_usage_cols$timestamp <- shiny_usage$started
+
+  static_usage <- get_usage_static(client, limit = Inf, from = from, to = to)
+  static_usage_cols <- static_usage[c("user_guid", "content_guid")]
+  static_usage_cols$timestamp <- static_usage$time
+
+  bind_rows(shiny_usage_cols, static_usage_cols)
+}
+
+get_usage <- function(client, from = NULL, to = NULL) {
```

Comment:
Here, but also elsewhere in these functions: you've got `from`/`to`, but are you using those anywhere here?

Reply:
(@jonkeane Just a side-note: I shared this function with @dethmasque `get_usage()` as it abstracts away the differences between the new firehose and older usage endpoints ([ref1](https://github.com/posit-dev/connect-extensions/pull/31), [ref2](https://github.com/posit-dev/connect-extensions/pull/31#discussion_r1985415947)). The `from` and `to` are there because in my extension, I decided to limit the query to the most recent week even before adding explicit date filters, for performance reasons.)